test/widgets/action_sheet_test.dart

```diff
@@ -1061,6 +1061,82 @@ void main() {
+        ZulipStream? channel,
+      }) async {
+        addTearDown(testBinding.reset);
+
+        final effectiveChannel = channel ?? eg.stream();
+        const topicName = TopicName('test topic');
+        final message = eg.streamMessage(stream: effectiveChannel, topic: 'test topic');
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+          realmUsers: [eg.selfUser],
+          streams: [effectiveChannel],
+          subscriptions: [eg.subscription(effectiveChannel)]));
+        store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+        connection = store.connection as FakeApiConnection;
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          foundOldest: true, messages: [message]).toJson());
+
+        await store.addMessage(message);
+        store.unreads.streams[effectiveChannel.streamId] ??= {};
+        store.unreads.streams[effectiveChannel.streamId]![topicName] ??= QueueList<int>();
+        store.unreads.streams[effectiveChannel.streamId]![topicName]!.add(message.id);
+
+        await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+          child: MessageListPage(initNarrow: TopicNarrow(effectiveChannel.streamId, topicName))));
+        await tester.pumpAndSettle();
+
+        await tester.longPress(find.byType(ZulipAppBar));
+        await tester.pump(const Duration(milliseconds: 250));
+      }
+
+      Future<void> setupToTopicActionSheetWithNoUnreadMessages(WidgetTester tester) async {
+        addTearDown(testBinding.reset);
+
+        final channel = eg.stream();
+        const topicName = TopicName('test topic');
+        final message = eg.streamMessage(stream: channel, topic: 'test topic', flags: [MessageFlag.read]);
+
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+          realmUsers: [eg.selfUser],
+          streams: [channel],
+          subscriptions: [eg.subscription(channel)]));
+        store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+        connection = store.connection as FakeApiConnection;
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          foundOldest: true, messages: [message]).toJson());
+
+        await store.addMessage(message);
+
+        await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+          child: MessageListPage(initNarrow: TopicNarrow(channel.streamId, topicName))));
+        await tester.pumpAndSettle();
+
+        await tester.longPress(find.byType(ZulipAppBar));
+        await tester.pump(const Duration(milliseconds: 250));
+      }
+
+      group('visibility', () {
+        testWidgets('shows button when topic has unread messages', (tester) async {
+          await setupToTopicActionSheetWithUnreadMessages(tester);
+
+          final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+          check(find.text(zulipLocalizations.actionSheetOptionMarkTopicAsRead)).findsOne();
```

Comment:
Let's use the string directly, without referencing `GlobalLocalizations`, because we don't need tests to respond to locale changes and that we can be clear about what is expected exactly.